iml-gui/crate/src/components/command_modal.rs

```diff
@@ -975,50 +974,39 @@ mod tests {
     fn test_async_handlers_consistency() {
         let mut rng = Xoroshiro64Star::seed_from_u64(485369);
         let db = build_db();
-
         let mut model = Model::default();
         let cmd_ids = db.all_cmds.iter().map(|x| x.id).collect::<Vec<_>>();
-        let mut permutation = vec![1, 2, 3];
-        let selects = vec![
-            Select::None,
-            Select::Command(2),
-            Select::Command(1),
-            Select::Command(3),
-            Select::Command(4),
-            Select::CommandJob(1, vec![10]),
-            Select::CommandJob(1, vec![10, 11]),
-            Select::CommandJobSteps(1, vec![10, 11], vec![21]),
-            Select::CommandJobSteps(1, vec![10, 11], vec![20, 21]),
-        ];
+        let selects = generate_random_selects(&db, &mut rng, 100);
         for select in selects {
-            let cmd_id = cmd_ids[rng.next_u32() as usize % cmd_ids.len()];
-            let (c, j, s) = prepare_subset(&db, cmd_id);
-            model.clear();
-            model.select = select.clone();
-            model.assign_commands(&c);
-            model.assign_jobs(&j);
-            model.assign_steps(&s);
-            let expected_cmd = model.commands_view.clone();
-            let expected_jobs = model.jobs_graph.clone();
-            let expected_steps = model.steps_view.clone();
-
-            model.clear();
-            model.select = select.clone();
-            shuffle(&mut permutation, &mut rng);
-
-            // we simulate, that FetchCommands, FetchJobs and FetchSteps come in arbitrary order
-            for p in &permutation {
-                match p {
-                    1 => model.assign_commands(&c),
-                    2 => model.assign_jobs(&j),
-                    3 => model.assign_steps(&s),
-                    _ => unreachable!(),
+            let permutations = vec![[1, 2, 3], [1, 3, 2], [2, 1, 3], [2, 3, 1], [3, 1, 2], [3, 2, 1]];
```

Comment:
You still can omit the first case (1, 2, 3), because it is the same. Or, strictly speaking, one of the six cases :)